provider/provider_test.go

```diff
@@ -25,8 +25,8 @@ func TestProvider(t *testing.T) {
 		})
 
 	select {
-	case <- time.Tick(1 * time.Second):
-		t.Fail()
+	case <- time.Tick(5 * time.Second):
```

Comment:
actually, why do we have time based test cases here? It's not very obvious for me. Maybe leave a comment here.